File: frontend/shop_kingkong/src/service/AuthService.js
import axios from "axios";

const axiosRaw = axios.create({
  baseURL: "http://localhost:8081/",
  withCredentials: true,
  headers: { "Content-Type": "application/json" },
});

const AuthService = {
  login(loginRequest) {
    return axiosRaw.post("auth/login", loginRequest);
  },
  logout(accountId) {
    return axiosRaw.post(`auth/logout/${accountId}`);
  },
  refresh() {
    return axiosRaw.post("auth/refresh");
  },
  resentOtp(email, action){
    return axiosRaw.post(`auth/resent?email=${email}&action=${action}`)
  }
};

export default AuthService;

File: backend/src/main/java/bag/service/auth/AuthService.java
package bag.service.auth;

import bag.modal.dto.AuthResponse;
import bag.modal.request.AuthRequest;

public interface AuthService {
    AuthResponse login(AuthRequest request);
    AuthResponse refreshToken(String refreshToken);
    void logout(int accountId);
    String resentOtp(String email, String action);
}
